vembrane/ann_types.py

```diff
@@ -140,12 +141,50 @@ def description(self):
 
 class AnnotationListEntry(AnnotationEntry):
     def __init__(
-        self, name: str, sep: str, typefunc: Callable[[str], Any] = None, **kwargs
+        self,
+        name: str,
+        sep: str,
+        typefunc: Callable[[str], Any] = None,
+        inner_typefunc: Callable[[str], Any] = lambda x: x,
+        **kwargs,
     ):
-        typefunc = typefunc if typefunc else lambda v: [x.strip() for x in v.split(sep)]
+        typefunc = (
+            typefunc
+            if typefunc
+            else lambda v: [inner_typefunc(x.strip()) for x in v.split(sep)]
+        )
         super().__init__(name, typefunc, nafunc=lambda: [], **kwargs)
 
 
+class AnnotationListDictEntry(AnnotationEntry):
+    def __init__(self, name: str, **kwargs):
+        def typefunc(x):
+            key_value_tuples = map(lambda v: v.split(":", maxsplit=1), x.split("&"))
+            mapping = defaultdict(list)
+            for key, value in key_value_tuples:
+                mapping[key].append(value)
+            # make sure non-existent keys do not return list() on __getitem__ calls
+            # in the filter expression (without actually building a new dict from
+            # the defaultdict)
+            mapping.default_factory = None
+            return mapping
+
+        super().__init__(name=name, typefunc=typefunc, **kwargs)
+
+
+class AnnotationNumberTotalEntry(AnnotationListEntry):
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name, sep="/", inner_typefunc=int, **kwargs)
+
+
+class AnnotationPredictionScoreEntry(AnnotationEntry):
```

Comment:
Is it intentional that these are lists of `int`/`float`? (Again, didn't look at the context here, sorry.)

Reply:
Yes, basically a pair `[number, total]`. Perhaps a custom type with number and total slots might be better?